From abe74be634c30ef3740aada980b2979629929cfb Mon Sep 17 00:00:00 2001
From: Yannick Fertre <yannick.fertre@foss.st.com>
Date: Tue, 4 Mar 2025 10:55:58 +0100
Subject: [PATCH] drm/stm: lvds: don't use pm_runtime

Pm_runtime should not be used when "pixel clock enable" is required.

Change-Id: I35da8e74ff8acfd3a48c42dfc5857e031a2e480f
Signed-off-by: Yannick Fertre <yannick.fertre@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/441601
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 drivers/gpu/drm/stm/lvds.c | 20 +++++++++++++-------
 1 file changed, 13 insertions(+), 7 deletions(-)

--- a/drivers/gpu/drm/stm/lvds.c
+++ b/drivers/gpu/drm/stm/lvds.c
@@ -557,12 +557,17 @@ static int lvds_pixel_clk_enable(struct
 	struct lvds_phy_info *phy;
 	int ret;
 
-	if (!pm_runtime_active(lvds->dev)) {
-		ret = pm_runtime_resume_and_get(lvds->dev);
-		if (ret < 0) {
-			DRM_ERROR("Failed to enable clocks, cannot resume pm\n");
-			return ret;
-		}
+	ret = clk_prepare_enable(lvds->pclk);
+	if (ret) {
+		DRM_ERROR("Failed to enable pclk: %d\n", ret);
+		return ret;
+	}
+
+	ret = clk_prepare_enable(lvds->pllref_clk);
+	if (ret) {
+		clk_disable_unprepare(lvds->pclk);
+		DRM_ERROR("Failed to enable pllref_clk: %d\n", ret);
+		return ret;
 	}
 
 	/* In case we are operating in dual link the second PHY is set before the primary PHY. */
@@ -626,7 +631,8 @@ static void lvds_pixel_clk_disable(struc
 			   PHY_GCR_DIV_RSTN | PHY_GCR_RSTZ);
 	}
 
-	pm_runtime_put(lvds->dev);
+	clk_disable_unprepare(lvds->pllref_clk);
+	clk_disable_unprepare(lvds->pclk);
 }
 
 static unsigned long lvds_pixel_clk_recalc_rate(struct clk_hw *hw,
